python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py

```diff
@@ -624,7 +624,10 @@ def inner(fn: Callable[..., Any]) -> AssetsDefinition:
                 )
 
             upstream_keys = {
-                dep for spec in specs for dep in spec.deps if dep not in output_tuples_by_asset_key
+                dep.asset_key
+                for spec in specs
+                for dep in spec.deps
+                # if dep.asset_key not in output_tuples_by_asset_key
```

Comment:
rm?

Reply:
yes - just confirming no tests fail because of this removal before fully deleting it